llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1306,6 +1307,12 @@ void PassBuilder::addVectorPasses(OptimizationLevel Level,
     FPM.addPass(BDCEPass());
   }
 
+  // We do UnrollAndJam in a separate LPM to Unroll ensure it happens first.
+  // In order for outer loop vectorization to be done, UnrollAndJam must occur before the SLPVectorizerPass.
+  if (EnableUnrollAndJam && PTO.LoopUnrolling) {
+    FPM.addPass(createFunctionToLoopPassAdaptor(
+        LoopUnrollAndJamPass(Level.getSpeedupLevel())));
+  }
   // Optimize parallel scalar instruction chains into SIMD instructions.
   if (PTO.SLPVectorization) {
```

Comment:
Is SLP vectorization on be default? If not then instead of moving UnJ, running another instance of the pass under this condition can be explored to avoid compile time penalties?

Reply:
Yeah, SLP is on by default, so unfortunately this wouldn't work because of the compile time penalty